src/js/edu-benefits/utils/validations.js

```diff
@@ -245,11 +245,19 @@ function isValidBenefitsInformationPage(data) {
   return data.chapter33 || data.chapter30 || data.chapter32 || data.chapter1606;
 }
 
+function isValidRelinquishedDate(field) {
+  // Allow dates up to two years ago
+  const pastDate = moment().subtract(2, 'years');
+  const date = dateToMoment(field);
+
+  return !isBlankDateField(field) && date.isAfter(pastDate);
```

Comment:
Will this have trouble with partially entered dates? You could check `date.isValid()`, too.